File: model/src/test/java/de/danoeh/antennapod/model/MediaTypeTest.java
package de.danoeh.antennapod.model;

import de.danoeh.antennapod.model.playback.MediaType;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class MediaTypeTest {

    @Test
    public void fromMimeTypeNullReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType(null));
    }

    @Test
    public void fromMimeTypeEmptyReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType(""));
    }

    @Test
    public void fromMimeTypeAudioPrefixReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("audio/mpeg"));
    }

    @Test
    public void fromMimeTypeVideoPrefixReturnsVideo() {
        assertEquals(MediaType.VIDEO, MediaType.fromMimeType("video/mp4"));
    }

    @Test
    public void fromMimeTypeApplicationOggReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/ogg"));
    }

    @Test
    public void fromMimeTypeApplicationOpusReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/opus"));
    }

    @Test
    public void fromMimeTypeApplicationXFlacReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/x-flac"));
    }

    @Test
    public void fromMimeTypeUnknownApplicationReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType("application/pdf"));
    }
}

File: model/src/test/java/de/danoeh/antennapod/model/MediaTypeTest.kt
package de.danoeh.antennapod.model

import de.danoeh.antennapod.model.playback.MediaType
import org.junit.Assert.assertEquals
import org.junit.Test

class MediaTypeTest {

    @Test
    fun fromMimeTypeNullReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType(null))
    }

    @Test
    fun fromMimeTypeEmptyReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType(""))
    }

    @Test
    fun fromMimeTypeAudioPrefixReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("audio/mpeg"))
    }

    @Test
    fun fromMimeTypeVideoPrefixReturnsVideo() {
        assertEquals(MediaType.VIDEO, MediaType.fromMimeType("video/mp4"))
    }

    @Test
    fun fromMimeTypeApplicationOggReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/ogg"))
    }

    @Test
    fun fromMimeTypeApplicationOpusReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/opus"))
    }

    @Test
    fun fromMimeTypeApplicationXFlacReturnsAudio() {
        assertEquals(MediaType.AUDIO, MediaType.fromMimeType("application/x-flac"))
    }

    @Test
    fun fromMimeTypeUnknownApplicationReturnsUnknown() {
        assertEquals(MediaType.UNKNOWN, MediaType.fromMimeType("application/pdf"))
    }
}
